instrumentation/github.com/Shopify/sarama/otelsarama/producer.go

```diff
@@ -35,9 +35,11 @@ type syncProducer struct {
 
 // SendMessage calls sarama.SyncProducer.SendMessage and traces the request.
 func (p *syncProducer) SendMessage(msg *sarama.ProducerMessage) (partition int32, offset int64, err error) {
-	span := startProducerSpan(p.cfg, p.saramaConfig.Version, msg)
+	span, ok := startProducerSpan(p.cfg, p.saramaConfig.Version, msg)
```

Comment:
I find the design with returning the `ok` is very error-prone and hard to read. 

Maybe it would be better if this if this `ok` would be somehow encapsulated in a type returned by `startProducerSpan` which then can be accepted by `finishProducerSpan` and used in `producerMessageContext` instead of `trace.Span`

Even this should be enough:
 
```
type producerSpan struct {
	span trace.Span
```

Then  the `finishProducerSpan` could even be implemented as `producerSpan`'s method

Reply:
What I originally wanted to do was just use a "noop" span and then avoid almost all conditionals, but I couldn't find anything like that in the Otel library.  I think what you are proposing is just a variation of a noop span, so I'll try that.

Reply:
ok @pellared take a look at my latest push and see what you think.